src/modules/document-caching/AmpDocumentCachablePlugin.ts

```diff
@@ -18,8 +18,20 @@
 import { Plugin } from 'workbox-cache-expiration';
 
 export default class AmpDocumentCachablePlugin extends Plugin {
-  constructor(config: any) {
-    super(config);
+  private allowedPages_: Array<RegExp>;
+
+  constructor(config: {
+    maxEntries?: Number;
+    maxAgeSeconds?: Number;
+    allowedNonAMPPages?: Array<RegExp>;
+  }) {
+    const { allowedNonAMPPages, ...pluginConfig } = config;
+    super(pluginConfig);
+    if (allowedNonAMPPages && Array.isArray(allowedNonAMPPages)) {
```

Comment:
Possible to make this early exit pattern instead of checking existance in successive condition checks?

```typescript
if (allowedNonAMPPages) {
  if (!Array.isArray(allowedNonAMPPages)) {
    throw new TypeError('allowedNonAMPPages should be an array of RegExp');
  }
  this.allowedPages_ = allowedNonAMPPages;
}
```